shell/common/mac/codesign_util.cc

```diff
@@ -105,6 +105,10 @@ bool ProcessSignatureIsSameWithCurrentApp(pid_t pid) {
   // Check whether the process meets the signature requirement of current app.
   status = SecCodeCheckValidity(process_code.get(), kSecCSDefaultFlags,
                                 self_requirement.get());
+  if (status == errSecCSUnsigned) {
+    // Process is not signed, no need to validate as it has full permission.
```

Comment:
If the parent process isn't signed but the app is IMO this should fail closed.  This may be an understandable error, but we shouldn't fail open a security check

Reply:
The check means to prevent a sandboxed app from stealing permissions for Electron apps, so if the parent process is unsigned then it already has full permissions and I think the check wouldn't be needed?